app/logic/Calendar/ICSProcessor.ts

```diff
@@ -28,85 +25,132 @@ export class ICSProcessor extends EMailProcessor {
 
 /* Find the iTIP method from a parsed vcalendar part */
 function iTIPMethod(ics: any): Scheduling {
-  switch (ics.vcalendar.method) {
+  switch (ics.VCALENDAR[0]?.METHOD?.[0].value) {
   case "CANCEL":
     return Scheduling.Cancellation;
   case "REQUEST":
     return Scheduling.Request;
   case "REPLY":
-    let vevent = Object.values(ics).find((event: any) => event.type == "VEVENT") as any;
-    switch (vevent?.attendee?.params?.PARTSTAT) {
-    case "DECLINED":
-      return Scheduling.Declined;
-    case "TENTATIVE":
-      return Scheduling.Tentative;
-    case "ACCEPTED":
-      return Scheduling.Accepted;
-    }
+    // Any other value will do, just make something up
+    return Scheduling.Declined + Scheduling.Tentative + Scheduling.Accepted;
   }
   return Scheduling.None;
 }
 
-function convertICSToEvent(ics: any): Event | null {
-  let vevent = Object.values(ics).find((event: any) => event.type == "VEVENT") as any;
-  if (!vevent) {
-    return null;
+declare global {
+  interface String {
+    toUpperCase<T extends string>(this: T): Uppercase<T>;
+  }
+}
+
+function unescaped(value: string): string {
+  return value.replace(/\\n|\\(.)/gi, (_, c) => c || "\n");
+}
+
+class icalEntry {
+  [s: Uppercase<string>]: string;
+  name: Uppercase<string>;
+  value: string;
+  line: string;
+  constructor(line: string) {
+    this.line = line;
+    let pos = line.search(/[;:]/);
+    this.name = line.slice(0, pos).toUpperCase();
+    line = line.slice(pos);
+    while (/^;([-\w]+)=("?)((\\?.)*?)\2(?=[;:])/.test(line)) {
+      this[RegExp.$1.toUpperCase()] = unescaped(RegExp.$3);
+      line = RegExp.rightContext;
+    }
+    this.value = unescaped(line.slice(1));
+  }
+}
+
+class icalContainer {
+  [s: Uppercase<string>]: icalEntry[];
+  parent: icalContainer | icalParser;
+  constructor(parent: icalContainer | icalParser) {
+    this.parent = parent;
   }
-  // Parser tries to simplify simple properties for us...
-  for (let prop in vevent) {
-    if (typeof vevent[prop] == "string") {
-      vevent[prop] = { params: {}, val: vevent[prop] };
+}
+
+class icalParser {
```

Comment:
Could you please put the parser (together with its aux classes/functions) in a new file ICalParser?
Also, class names should be `ICalParser` and `ICalEntry`, because all classes must be Capitalized.